### ISSUE TITLE
Stange circles in dxf line work

### ISSUE THREAD

**Issue by katSchmid** [gdsestimating/three-dxf#64]
hi,
Unfortunately i cannot share the file but we have some linework showing strange circles instead of buldges (left: ours, right three-dxf)
![image](https://user-images.githubusercontent.com/65633379/128303847-bdc613b1-1f71-492e-8804-bbd476631a94.png)
and qgis for comparison
![image](https://user-images.githubusercontent.com/65633379/128303917-e1810559-df06-4d5d-bd7f-42e08ea1cc03.png)

**Comment by Serthys** [comment]
Can you copy a small portion of that file containing one of those buldges into a new file and share that?

**Comment by katSchmid** [comment]
i will try, has three-dxf an easy way to split out curves?

**Comment by Serthys** [comment]
I did a quick test with Splines and there are some strange things, I'm not sure if it's related to your problems.
Sample file: [spline test.zip](https://github.com/gdsestimating/three-dxf/files/6944984/spline.test.zip)
This is how it should look like:
![image](https://user-images.githubusercontent.com/13573099/128498723-bb295b7f-6840-4ecb-b707-988403a2ccf3.png)
This is how it looks:
![image](https://user-images.githubusercontent.com/13573099/128498905-4694ef9b-6692-4f05-99b9-f299fc62ab47.png)

The odd curves are not drawn correctly and where the spline has an odd number or points it joins the last one with 0,0,0.

**Comment by katSchmid** [comment]
I dont have a real dxf editor, can i jsut dm you the file?

**Comment by bzuillsmith** [comment]
@Serthys yeah, I didn't really know what I was doing when I implemented splines and I didn't have a good program to test things with at the time. It looks like you created a nice test for those. Would love a PR, even if it's just to add a failing test.

The issues here definitely look like issues with bulges. Also all the bulges seem to be originating from the end of a line so perhaps we aren't handling some edge condition for bulges? We'll definitely need at least a portion of the file to diagnose. Are you able to open the file and remove most of it but leave part of what is broken? Then you would just be posting a very tiny piece? Otherwise I suppose a DM will do.

**Comment by bzuillsmith** [comment]
@Serthys I just pushed a partial fix for splines. It will now draw simple ones, but can't handle more that 4 control points. Three.js doesn't have an algorithm for more than that. I understand the fundamentals of splines much better now, but not well enough to write code that can handle any number of control points. From what I understand, it becomes a recursive computation as you add more points.

I plan to take a look at the bulge problem soon. Sorry, been getting reoriented with everything. It's been a while!
